src/display.rs

```diff
@@ -177,8 +178,10 @@ impl Display {
         let cell_height = (metrics.line_height + font.offset().y() as f64) as u32;
 
         // Resize window to specified dimensions
-        let width = cell_width * options.columns_u32() + 4;
-        let height = cell_height * options.lines_u32() + 4;
+        let columns = options.columns().unwrap_or_else(|| dimensions.columns());
+        let lines = options.lines().unwrap_or_else(|| dimensions.lines());
```

Comment:
With the changes to options/config/dimensions discussed elsewhere, this can become

```rust
let dimensions = options.dimensions()
    .unwrap_or_else(|| config.dimensions());
```

Then `lines` and `columns` can be extracted from `dimensions`.